runtime/BUILD.gn

```diff
@@ -74,81 +72,105 @@ source_set("runtime") {
     "window_data.h",
   ]
 
+  public_deps = [
+    "//third_party/rapidjson",
+  ]
+
+  public_configs = [ "//flutter:config" ]
+
   deps = [
     ":test_font",
     "//flutter/assets",
     "//flutter/common",
-    "//flutter/flow",
     "//flutter/fml",
     "//flutter/lib/io",
-    "//flutter/lib/ui",
     "//flutter/third_party/tonic",
     "//flutter/third_party/txt",
     "//third_party/dart/runtime:dart_api",
     "//third_party/dart/runtime/bin:dart_io_api",
     "//third_party/skia",
   ]
 
-  public_deps = [
-    "//third_party/rapidjson",
-  ]
-
-  public_configs = [ "//flutter:config" ]
-
   if (flutter_runtime_mode != "release" && !is_fuchsia) {
     # Only link in Observatory in non-release modes on non-Fuchsia. Fuchsia
     # instead puts Observatory into the runner's package.
     deps += [
       "//third_party/dart/runtime/observatory:embedded_observatory_archive",
     ]
   }
+
+  deps_legacy_and_next = [
+    "//flutter/flow:flow",
+    "//flutter/lib/ui:ui",
+  ]
 }
 
 test_fixtures("runtime_fixtures") {
   dart_main = "fixtures/runtime_test.dart"
 }
 
-source_set("runtime_unittests_common") {
-  visibility = [ ":*" ]
-
+source_set_maybe_fuchsia_legacy("runtime_unittests_common") {
   testonly = true
 
   sources = [
+    "dart_isolate_unittests.cc",
+    "dart_lifecycle_unittests.cc",
+    "dart_service_isolate_unittests.cc",
+    "dart_vm_unittests.cc",
     "runtime_test.cc",
     "runtime_test.h",
   ]
 
   public_deps = [
     ":libdart",
-    ":runtime",
     ":runtime_fixtures",
     "//flutter/common",
     "//flutter/fml",
     "//flutter/lib/snapshot",
-    "//flutter/shell/common",
+
+    #"//flutter/shell/common",
```

Comment:
remove this or keep it?

Reply:
Looks like this was a false dependency (as suspected when I commented it out), so I can likely remove it